### PR TITLE
OCPQE-16217 Patch all admin acks

### PR DESCRIPTION
**Pull request by JianLi-RH**
/cc @jiajliu @shellyyang1989

### CODE REVIEW

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.11-aws-ipi-f14

**Comment by shellyyang1989** [comment]
@JianLi-RH what admin gates do you see in the failed job? I'm wondering why we have multiple gates not acked.

**Comment by JianLi-RH** [comment]
@shellyyang1989 We don't see multiple admin gates, creating this PR is just for debug...

**Comment by JianLi-RH** [comment]
@shellyyang1989 looks like there are really 2 gates: https://storage.googleapis.com/qe-private-deck/pr-logs/pull/openshift_release/41994/rehearse-41994-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.11-aws-ipi-f14/1687319639605907456/build-log.txt?Expires=1691133372&GoogleAccessId=qe-private-deck%40openshift-ci-private.iam.gserviceaccount.com&Signature=DH6QaAkKJevh6nVpyk1qCl%2B%2FZoEEFraQ4XARHp6aRk1fNT8GFuoo8nJHD6zydz9f9f4PoptPb%2Fi3VsusKVndwJ1ujhpQsf%2B3eDHl1pXPivusLBDRtYWipw3Lq8w5NuQ%2FWM7bNA0IexV1e%2BmdN5yMSRKLLSlXwxYDcY7i%2F9xe1BPoIw2H4ei%2B2s7En%2FMpBDB10x%2B3rx8zcOdzAUYOElKEpr2HQhRRL0PAB7bgphPBC%2Bx1wsPlqaO4%2B9ij1KNLDEe61dhuG8veaRjBoUFnnorY23nz%2B3sSz1PArI0i%2BcxJzqS%2Bz8b3uB01AfZZIgfSLCNqDGK4Hh040i%2F3c2%2FlRG1n1g%3D%3D

**Comment by shellyyang1989** [comment]
Yeah, imho, it makes sense to have 2 gates here but "ack-4.11-kube-1.25-api-removals-in-4.12" has been acked, so it should not gate 4.12 to 4.13 upgrade any more. Ack of "ack-4.12-kube-1.26-api-removals-in-4.13" should be able to unblock upgrade.

**Comment by JianLi-RH** [comment]
@shellyyang1989 @jiajliu so previously our script always get the first gate to patch and it makes patching not finished, here is the old log, you can find there are 2 lines of **Admin ack patch data is: ack-4.11-kube-1.25-api-removals-in-4.12**
https://storage.googleapis.com/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.11-aws-ipi-f14/1683468060184285184/build-log.txt?Expires=1691112141&GoogleAccessId=qe-private-deck%40openshift-ci-private.iam.gserviceaccount.com&Signature=R%2BbCnoYQnfFmYHM8iOS7f44yP0E5maizyatRPWq07BRReYpUDw6n7i2nduo1Mc3%2BSxXumSGYArj%2FEktIZxvh%2BKxrub29VG7mQKkdooG0n7VlRw7gvF7NFVPMYmc8wZbzcokbeTTcoEhFfxUlZ5890kQmnyl4jiFPjjWZM2i57QEYbx5eug98p1VMPJdkdRNTmt%2BrgGGHwTFhbjX%2F%2BxnGPssbM7DPjr87%2FCCshvhOK6pbRV75ghECO7KQx6ANNgaFYtTDLdJjaZF%2BjI%2BlZ4bVDh0Lm8dCFJOaImOH057nJyh45AdaGMZqNPUmfAGJl9D2k2aM2pjOdlUbb1bl8N%2BdhA%3D%3D

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.11-aws-ipi-f14

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.11-aws-ipi-f14

**Comment by shellyyang1989** [comment]
I'm afraid patching gates that have acked before would miss some check points. One option is to patch the gate matching the name pattern `api-removals-in-<target>`

**Comment by jiajliu** [comment]
@JianLi-RH You're right, the root cause is that it getting a wrong key during 2nd upgrade. We meant no need to patch all admin gates in each round upgrade, because upgrade is a loop in chain upgrade, and so we only need add ack for the correct one in each upgrade loop.

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.11-aws-ipi-f14

**Comment by JianLi-RH** [comment]
The issue is solved: https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/41994/rehearse-41994-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.11-aws-ipi-f14/1687631602030481408/artifacts/aws-ipi-f14/cucushift-chainupgrade-toimage/build-log.txt

you can search `All admin acks:` to check admin ack.

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.11-aws-ipi-f14

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.11-aws-ipi-f14

**Comment by JianLi-RH** [comment]
@shellyyang1989 updated the if condition as you suggested and the test passed: https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/41994/rehearse-41994-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.11-aws-ipi-f14/1688514697004847104/artifacts/aws-ipi-f14/cucushift-chainupgrade-toimage/build-log.txt

File: ci-operator/step-registry/cucushift/chainupgrade/toimage/cucushift-chainupgrade-toimage-commands.sh [hunk 1]
```diff
@@ -414,18 +414,27 @@ function admin_ack() {
...
     echo "Require admin ack"
     local wait_time_loop_var=0 ack_data
-    ack_data="$(echo ${out} | awk '{print $2}' | cut -f2 -d\")" && echo "Admin ack patch data is: ${ack_data}"
-    oc -n openshift-config patch configmap admin-acks --patch '{"data":{"'"${ack_data}"'": "true"}}' --type=merge
+    ack_data="$(echo ${out} | jq -r "keys[]")" && echo "Admin ack patch data is: ${ack_data}"
+    for ack in ${ack_data};
+    do
+        # e.g.: ack-4.12-kube-1.26-api-removals-in-4.13
+        if [[ "${ack}" == *4\.$TARGET_MINOR_VERSION ]] 
+        then
```
Comment (jiajliu) [inline ci-operator/step-registry/cucushift/chainupgrade/toimage/cucushift-chainupgrade-toimage-commands.sh:429]: A tiny suggestion here, how about move the output here? No need all keys `echo "Admin ack patch data is: ${ack_data}"` in line 424, but ${ack} is the correct one which we care about in output.

Comment (JianLi-RH) [reply ci-operator/step-registry/cucushift/chainupgrade/toimage/cucushift-chainupgrade-toimage-commands.sh:429]: `echo "Admin ack patch data is: ${ack_data}"` helps us understand how many gates we have, I think it is useful for debug.

and we can also output the exact one. up to you

Comment (JianLi-RH) [reply ci-operator/step-registry/cucushift/chainupgrade/toimage/cucushift-chainupgrade-toimage-commands.sh:429]: good catch, updated @jiajliu

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.11-aws-ipi-f14

**Comment by jiajliu** [comment]
test pass:
```
INFO[2023-08-08T13:08:09Z] Running step aws-ipi-f14-cucushift-chainupgrade-toimage. 
INFO[2023-08-08T15:52:25Z] Step aws-ipi-f14-cucushift-chainupgrade-toimage succeeded after 2h44m16s.
...
INFO[2023-08-08T17:02:54Z] Running step aws-ipi-f14-cucushift-upgrade-toimage. 
INFO[2023-08-08T18:22:01Z] Step aws-ipi-f14-cucushift-upgrade-toimage succeeded after 1h19m6s.
```

**Pull request merged**